packages/react-utilities/src/hooks/useFirstMount.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+
+/**
+ * Checks if components was mounted the first time.
+ * @example
+ * const isFirstMount = useFirstMount();
+ */
+export function useFirstMount(): boolean {
+  const isFirst = React.useRef(true);
+
+  if (isFirst.current) {
```

Comment:
IIRC this ought to be run inside `useEffect` in case of async rendering which may call component renders multiple times in the same cycle

Reply:
But the subsequent renders are after the first mount ?

Reply:
Not necessarily, in concurrent mode (that was what I meant to say instead of "async") it's possible that render methods and certain class component lifecycles may get called multiple times within a single render cycle. This is the best explanation I can find in the docs. https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects

I'm pretty sure there's also guidance on wrapping `ref.current` updates in `useEffect` but I'm having trouble finding this, though most of the examples in the React docs do it this way.

Reply:
Here's an article explaining in more detail, though it's not an official source.
https://betterprogramming.pub/how-to-properly-use-the-react-useref-hook-in-concurrent-mode-38c54543857b

Reply:
Leaving as is, since according to this codesandbox, I cannot repro the multi rerender behaviour on first mount, it pretty consistently renders only once on first mount:

https://codesandbox.io/s/heuristic-brook-s4w0q?file=/src/App.jsx

Reply:
Will add a comment to potentially reverify